modules/database/src/ioc/db/dbEvent.c

```diff
@@ -1009,7 +1011,7 @@ static int event_read ( struct event_que *ev_que )
             if (pfl) {
                 /* Issue user callback */
                 ( *user_sub ) ( pevent->user_arg, pevent->chan,
-                                ev_que->evque[ev_que->getix] != EVENTQEMPTY, pfl );
+                                eventsRemaining, pfl );
```

Comment:
Looking again at the previous code.  `ev_que->evque[ev_que->getix]` is being evaluated without locking, so a race condition as well.